mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksFactory.kt

```diff
@@ -43,16 +43,19 @@ private fun <VM : MavericksViewModel<S>, S : MavericksState> createViewModel(
     val viewModelClass = stateRestorer?.viewModelClass ?: declaredViewModelClass
     val stateClass = stateRestorer?.stateClass ?: declaredStateClass
 
-    val factoryViewModel = viewModelClass.factoryCompanion()?.let { factoryClass ->
```

Comment:
Could we first try the declared viewModelClass / StateClass for a no-args constructor?

We then only use the persisted state class in the event the declared class fails. I think it achieves the same result as here, but seems like a bit safe change, as it only changes a case that would have failed before.

Reply:
For example, I think this this approach prioritizes the most concrete factory. So if we had BaseVM.Factory and ChildVM.Factory - and BaseVMFactory produces a ChildVM, then in the restore case ChildVM would be called.

Alternatively, what we need to record in the saved state is the VM class that we _actually_ invoked to create the state. 

Stepping back -- the real goal is to record whatever VM/S class was used to create the view model so we can do the same thing in recreate.